api/client/cfg.py

```diff
@@ -6,3 +6,4 @@
 MAX_QUERIES_PER_SECOND = 10
 TIMEOUT=6000
 MAX_RESULT_COMBINATION_DEPTH=3
+MAX_SERIES_PER_COMB=1000
```

Comment:
With this as 1000, I was finding that "production", "corn", "united states" was taking ~5 minutes to run. Killed it prematurely a few times thinking it was unresponsive before I added some logs and realized it was just looping through thousands of results

Reduced this number down to 5, and the same query only takes 5 seconds. And I got the same top result

It's a trade-off between hit ratio and responsiveness, but 1000 seems really high to me

Reply:
i've been using a lot of examples (6 or 7 crops, 3 countries, districts and provinces).. 100 is too small (especially when you have  e.g a metric= land cover, item = a crop that has a lot of varieties, region = a region with many descendants that have different items, e.g. millet. 
US corn being one of the biggest makes sense, obviously there are  quite a few  items and many many region levels to consider.

 its a trade-off but the purpose of find_data_series is deep exploration so the tradeoff is very biased for quality than speed. the performance sensitive operations are getting data.  
in terms of this PR scope, let's stick to going rfom unlimited to 1000.

Reply:
addressed with generator impl below